taxonomy/validators/program.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Validator for course metadata provider.
+
+All host platform must run this validator to make sure providers are working as expected.
+"""
+from taxonomy.providers.utils import get_program_provider
+
+
+class ProgramProviderValidator:
+    """
+    Validate that the interface requirement for program provider matches with the implementation.
+    """
+
+    def __init__(self, test_programs):
+        """
+        Setup an instance of program provider.
+
+        Since, these tests will run inside host application, list of test courses will also provided by the host.
+        Args:
+            test_programs (list<str>): List of program UUIDs in the form of string.
+        """
+        self.test_programs = test_programs
+        self.program_provider = get_program_provider()
```

Comment:
I don't understand the addition of test_programs here.

Reply:
Here in test_validator 
https://github.com/openedx/taxonomy-connector/pull/98/files#diff-6ab916975a15e776ca67e412324892450797a072021e31b5485e82172e167b04R47

We are sending test_programs uuid to validate given programs

Reply:
So this is used only in unit tests? If yes, we can change its location and move the file to tests directory.

Reply:
Actually this validator is only for testing and validate the data of the Program or Course. So as per current structure this is written in validators. Unit tests are written separately which will use this validator to validate data

Reply:
Right. Ideally, it is better to keep testing utils under tests directory. For now, we can go ahead. But this should be refactored in the future. Seeing a test util outside tests directory causes confusion

Reply:
We also have test utils, but here let say we want to validate Program id (1,2,3,4,5). Then we will call this validator by passing these ids in __init__ and it will return either these are valid or not. If we move these to tests directory then we can't send our program ids to validate them. 
Does it make sense?

Reply:
These IDs are from Discovery and not tests, right?

Reply:
These IDs are from taxonomy tests and discovery tests as well. reference: https://github.com/openedx/course-discovery/blob/master/course_discovery/apps/taxonomy_support/tests/test_providers.py#L40